src/snakemake/sourcecache.py

```
@@ -29,6 +28,10 @@
 from snakemake.logging import logger
 
 
+if TYPE_CHECKING:
```

Ruff (0.11.9)

Undefined name `TYPE_CHECKING`

(F821)